Firebase/Auth/Source/FIRAuth.m

```diff
@@ -857,12 +857,6 @@ - (void)internalSignInAndRetrieveDataWithCredential:(FIRAuthCredential *)credent
                                        requestConfiguration:_requestConfiguration];
   request.autoCreate = !isReauthentication;
   [credential prepareVerifyAssertionRequest:request];
-  if ([credential isKindOfClass:[FIROAuthCredential class]]) {
```

Comment:
just to let reviewer know --- this is moved to FIROAuthCredential's prepareVerifyAssertionRequest: